# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasession/MediaSessionUtils.h
# Commit: a008d3cebfdb
# Full Hash: a008d3cebfdb9d96ab32530b90180141575316a8
# Author: alwu <alwu@mozilla.com>
# Date: 2020-03-17 09:36:40
# Regressor Bug: 1582508
# File Overlap Count: 1
# Description:
#   Bug 1582508 - part4 : propagate declared playback state to media session controller. r=chunmin
#   
#   This patch includes the implementation of propagating declared playback state from the media session in the content process to the media session controller in the chrome process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66341
# ==============================================================================

diff -r a73415e9c006 -r a008d3cebfdb dom/media/mediasession/MediaSessionUtils.h
--- a/dom/media/mediasession/MediaSessionUtils.h	Tue Mar 17 00:03:44 2020 +0000
+++ b/dom/media/mediasession/MediaSessionUtils.h	Tue Mar 17 00:17:36 2020 +0000
@@ -33,6 +33,21 @@
   }
 }
 
+inline const char* ToMediaSessionPlaybackStateStr(
+    const MediaSessionPlaybackState& aState) {
+  switch (aState) {
+    case MediaSessionPlaybackState::None:
+      return "none";
+    case MediaSessionPlaybackState::Paused:
+      return "paused";
+    case MediaSessionPlaybackState::Playing:
+      return "playing";
+    default:
+      MOZ_ASSERT_UNREACHABLE("Invalid MediaSessionPlaybackState.");
+      return "Unknown";
+  }
+}
+
 }  // namespace dom
 }  // namespace mozilla
 
